File: app/src/main/java/com/divercity/app/features/profile/selectmajor/major/MajorPaginatedRepository.java
package com.divercity.app.features.profile.selectmajor.major;

import android.support.annotation.Nullable;

import com.divercity.app.core.utils.Listing;
import com.divercity.app.data.entity.major.MajorResponse;

/**
 * Created by lucas on 01/10/2018.
 */

public interface MajorPaginatedRepository {

    Listing<MajorResponse> fetchMajors(@Nullable String query);

    void retry();

    void refresh();

    void clear();

}

File: app/src/main/java/com/divercity/android/features/onboarding/selectmajor/MajorPaginatedRepository.kt
package com.divercity.android.features.onboarding.selectmajor

import com.divercity.android.core.base.datasource.BaseDataSourceRepository
import com.divercity.android.core.base.usecase.Params
import com.divercity.android.core.base.usecase.UseCase
import com.divercity.android.data.entity.major.MajorResponse
import com.divercity.android.features.onboarding.selectmajor.usecase.FetchMajorsUseCase
import javax.inject.Inject

/**
 * Created by lucas on 01/10/2018.
 */

class MajorPaginatedRepository @Inject
internal constructor(private val fetchMajorsUseCase: FetchMajorsUseCase) :
    BaseDataSourceRepository<MajorResponse>() {

    override fun getUseCase(): UseCase<List<MajorResponse>, Params> = fetchMajorsUseCase
}
